src/site/content/en/angular/precaching-with-the-angular-service-worker/index.md

```diff
@@ -0,0 +1,103 @@
+hero: performance-budgets-with-the-angular-cli.jpg
+date: 2019-07-02
+description: |
+  Learn how to use the Angular service worker to precache the static assets in your app.
+authors:
+  - mgechev
+---
+
+## Dealing with limited connectivity
+
+When users have limited network access—or none at all—web app functionality can significantly degrade and often fails. Using a [service worker](https://developers.google.com/web/fundamentals/primers/service-workers/) to provide precaching lets you intercept network requests and deliver responses directly from a local cache instead of retrieving them from the network. Once your app’s assets have been cached, this approach can really speed up an app and make it work when the user is offline.
+
+This post walks through how to set up precaching in an Angular app. It assumes you’re already familiar with precaching and service workers in general. If you need a refresher, check out the “[Service workers and the Cache Storage API](https://web.dev/service-workers-cache-storage/)” post.
+
+_You can find the code for the current example [on GitHub](https://github.com/mgechev/service-worker-web-dev)._
+
+## Introducing the Angular service worker
+
+The Angular team offers a service worker module with precaching functionality that’s well integrated with the framework and the [Angular command line interface (CLI)](https://cli.angular.io/).
+
+Here’s a basic Angular app:
+
+<div class="glitch-embed-wrap" style="height: 480px; width: 100%;">
+  <iframe
+    src="https://stackblitz.com/github/mgechev/service-worker-web-dev/tree/external-image?embed=1&file=src/app/app.component.ts&view=preview"
+    style="height: 100%; width: 100%; border: 0;">
+  </iframe>
+</div>
+
+To add the service worker:
+
+1. Run this command in the CLI:
+
+```
+$ ng add @angular/pwa
+```
+
+{% Aside %}
+
+If you have multiple projects in the Angular CLI workspace, you can optionally specify a `--project` property with the project name you want to add the service worker to.
+
+{% endAside %}
+
+`@angular/service-worker` and `@angular/pwa` should now be installed in the app and should appear in `package.json`. The `ng-add` [schematic](https://angular.io/guide/schematics) also adds a file called `ngsw-config.json` to the project, which you can use to configure the service worker. (The file includes a default configuration that you’ll customize a little later.)
+
+Now build the project for production:
+
+```
+$ ng build --prod
+```
+
+Inside the `dist/[PROJECT_NAME]` directory you’ll find a file called `ngsw.json`. This file tells the Angular service worker how to cache the assets in the app. The file is generated during the build process based on the configuration (`ngsw-config.json`) and the assets produced at build time.
+
+Now start an HTTP server in the directory containing your app’s production assets, open the public URL, and check out its network requests in Chrome DevTools:
+
+{% Instruction 'preview' %}
```

Comment:
This is the shortcode for full-screen view in Glitch. It needs to be replaced with instructions for how to open a new window in StackBlitz.

Reply:
Since StackBlitz was abandoned, this instruction needs to be update to match what readers will do with the GH version.